server/test/controllers/device/device.controller.test.js

```diff
@@ -107,6 +107,38 @@ describe('GET /api/v1/device_feature/aggregated_states', () => {
   });
 });
 
+describe('GET /api/v1/device_feature/:device_feature_selector/states', () => {
```

Comment:
Overall, same feedbacks as before.

This test files looks very complex for a very simple feature.

It's complex because it's a copy paste from another test suite that had a complete different purpose, I think it might be great to rewrite those tests in a way simpler way.

Reply:
Hello @Pierre-Gilles,
Indeed it is mainly a copy/paste, but with modifications. The basis of the "states" query being the same. I've read, mostly understood, and edited accordingly to the need I thought (most certainly visibly wrong).
I thought that we really had to test everything to be sure that in the future, even if by mistake a modification of the DB or other happened, we could transcribe it. But in fact I certainly do too much.
So I reduced to the best of my understanding on your vision. Including property checks in this file.
In the file `server/test/lib/device/device.getDeviceFeaturesStates.test.js` I removed the 1st test which effectively is the repetition of this one.
> Pierre-Gilles 2 days ago
> Overall, same feedback as before.

However for this part, I don't see how to put the date in hard since I use the `insertStates` already created in line 10 which uses the date and time of the day. I was able to perform the tests with data for November / December, it did not pose any problems.

> It's complex because it's a copy paste from another test suite that had a complete different purpose, I think it might be great to rewrite those tests in a simpler way.

For this part, I'm sorry, the writing of the tests being already very complex for me, I don't see how to simplify these more after this revision. If that still doesn't fit, do you want me to ask Alex as well if he has some time.